### DISCUSSION TITLE
More Dynamic Movelists - RemapMovelist

### DISCUSSION THREAD

**Discussion by RealFoobs** [ikemen-engine/Ikemen-GO#1556] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

In many commercial games you can play extra versions of existing characters which may or may not have radical toolset changes, and the in-game movelists pretty much always reflects this by swapping to a different command list. Games with characters that can transform in the middle of a fight (probably?) operate on the same logic.

In ikemen (and mugen before it) having 2 or even more incarnations of a same character within a single work is commonplace, usually tying different modes to palno or a pre-intro selector, but unless one uses different .defs for each character, there's no way to change movelist.dat on the fly. Of course one could simply add all possible moves and leave notes on this file, but that can render the movelist too verbose and less useful.

I imagine incorporating the full game logic in the existing movelist format taken from mame might be colossal wrok for very little payoff, so what about limiting it to a single state controller: RemapMovelist.

### Describe the solution you'd like

The idea here is that one would define multiple movelist parameters in the character's def, and then one would change them as needed. So for exampple for a Ryu that has an Evil Ryu mode accesible by holding start in the select screen, one would do something like this:

in Ryu.def

```
[Files]
[...]
movelist = ryu.dat
movelist1 = evilryu.dat
```
and in state 5900

```
[StateDef 5900; type: S;]
if palno > 6 {remapMovelist{value:1}}
```

### Search terms

movelist

**Comment by K4thos** [comment, marked as answer]
implemented #3608
